cmake/CheckAtomic.cmake

```diff
@@ -0,0 +1,43 @@
+# - Try to find if 64-bits atomics need -latomic linking
+# Once done this will define
+#  HAVE_CXX_ATOMICS_WITHOUT_LIB - Whether atomic types work without -latomic
+
+include(CheckCXXSourceCompiles)
+include(CheckLibraryExists)
+
+# Sometimes linking against libatomic is required for atomic ops, if
+# the platform doesn't support lock-free atomics.
+
+function(check_working_cxx_atomics VARNAME)
+check_cxx_source_compiles("
```

Comment:
Please indent this call (0 -> 4) and `message()` call below (2 -> 4). Also remove space between command name and opening paren in `if ()` below. I know, I know, we still need to add a CMake linter at some point... Otherwise LGTM.